src/test/java/edu/hm/hafner/util/SetTest.java

```diff
@@ -0,0 +1,162 @@
+package edu.hm.hafner.util;
+
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.Set;
+
+import org.junit.jupiter.api.Test;
+
+import static org.assertj.core.api.Assertions.*;
+
+/**
+ * Verifies that {@code HashSet<Integer>} objects comply with the contract in {@link HashSet} and {@link Set}.
+ *
+ * @author Marc Hennig
+ */
+public class SetTest {
+    @Test
+    void shouldHandleInvalidConstruction() {
+        assertThatExceptionOfType(NullPointerException.class).isThrownBy(() -> new HashSet<Integer>(null));
```

Comment:
Diesen Konstruktor sollte man auch mit einer gefüllten Collection testen.

Reply:
Das wird in späteren Testcases doch überprüft - wäre es notwendig das der From halber doppelt durchzuführen?

Reply:
Implizite Tests oder Tests die mehrere Sachen Testen sind meistens schwer nachzuvollziehen, wenn mal ein Test fehl schlägt. Es kann natürlich in manchen Situationen auch sinnvoll sein und wie vieles beim Testen ist es auch ein bisschen Geschmackssache. Es würde also der Form halber nicht schaden.

Reply:
Auch nachzuvollziehen was alles getestet wird, ist durch Implizite Tests schwer.